mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -862,6 +862,70 @@ def get_metric_history(self, run_id, metric_key, max_results=None, page_token=No
             metrics = session.query(SqlMetric).filter_by(run_uuid=run_id, key=metric_key).all()
             return PagedList([metric.to_mlflow_entity() for metric in metrics], None)
 
+    class MetricWithRunId(Metric):
+        def __init__(self, run_id, metric: Metric):
```

Comment:
```suggestion
        def __init__(self, metric: Metric, run_id):
```

Reply:
Nit, to reflect class name and information hierarchy

Reply:
Good point! Committed!